### PR TITLE
Use db functions instead of swap-tx!

### PR DESCRIPTION
**Pull request by ericnormand**
I was having lots of trouble reproducing a heisenbug, so I rewrote the
swap-tx! implementation as a db function. It turns out to be much
simpler. The tests are easier as well.

I brought in test.check to try to reproduce the old heisenbug and I
think it's still useful for verifying certain properties of parallel
execution. Namely, I'm verifying that we don't get into an illegal
state.

I haven't been able to produce any bugs even with pretty aggressive
usage of parallel threads banging on the db,  including with an external
dev datomic transactor. I'm confident this works.

My main concern is that this data model does not fit well into its
usage. I would much rather see it somewhat inverted: ocd-division at the
top, with subcomponents for registration and election.

### CODE REVIEW

**Comment by ericnormand** [comment]
[Pivotal Card](https://www.pivotaltracker.com/story/show/108985506)

**Comment by danielglauser** [comment]
@ericnormand Is there an easy way to run the parallel generative tests against the old `swap-tx` fn to see if we can reproduce the original issue? Overall I think that the new code is simpler and is an easier way to avoid race conditions so I'm happy about it.

**Comment by ericnormand** [comment]
@danielglauser Yes. But reproducing them is not so easy. I have to run them several times before I get them. Just run the tests on master.

File: src/election_authority_works/entities/authority.clj [hunk 1]
```diff
@@ -211,134 +211,63 @@
...
+    (:office-name authority)
+
+    :authority/official-title
+    (:official-title authority)
+
+    :authority/type
+    (:types authority)
+
+    :authority/physical-address
+    (some-> authority
```
Comment (danielglauser) [inline src/election_authority_works/entities/authority.clj]: Nice use of `some->`

File: resources/schemas/20151125-db-functions.edn [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                         db authority-id)]
+           (for [[eid] eids]
+             [:db.fn/retractEntity eid]))}}
+
+ {:db/id #db/id [:db.part/user]
+  :db/ident :authority/upsert
+  :db/fn #db/fn
+  {:lang "clojure"
+   :params [db authority]
+   :code (let [one-type? (fn [e] (= 1 (count (:authority/type e))))
```
Comment (danielglauser) [inline resources/schemas/20151125-db-functions.edn]: This is really nicely laid out. Well done sir!

**Comment by danielglauser** [comment]
LGTM

**Pull request merged**